client/src/quic_client.rs

```diff
@@ -66,6 +67,7 @@ struct QuicClient {
 
 pub struct QuicTpuConnection {
     client: Arc<QuicClient>,
+    connection_stats: Arc<ConnectionCacheStats>,
```

Comment:
sticking a cache implementation detail into the connection doesn't feel great.  can you break out this refactor to it's own commit so we can consider it in isolation?

not having all of the plumbing changes to sift through will make review the rest easier too

Reply:
This was done in another PR.